payment/controllers.go

```diff
@@ -276,10 +233,69 @@ func GetOrder(service *Service) handlers.AppHandler {
 			status = http.StatusNotFound
 		}
 
+		if order != nil && !order.IsPaid() && order.IsStripePayable() {
```

Comment:
this whole conditional seems like a huge hack. why do we have a separate `OrderWithStripeCheckoutSessionID` type? shouldn't we just fetch and return the metadata as part of `GetOrder`?